doc/rst/language/spec/statements.rst

```diff
@@ -699,6 +700,65 @@ A ``break`` statement cannot be used to exit a parallel loop
         }
       }
 
+.. _The_Require_statement:
+
+The Require Statement
+---------------------
+
+The require statement provides a means to specify required files from
+within the program. It has an effect similar to adding the specified
+files to the Chapel compiler's command line. The filenames are relative
+to the directory from which the Chapel compiler was invoked. Any
+directories specified using the Chapel compiler's -I or -L flags will
+also be searched for matching files.
+
+.. code-block:: syntax
+
+   require-statement:
+     'require' string-or-identifier-list
+
+   string-or-identifier-list:
+     string-or-identifier
+     string-or-identifier ',' string-or-identifier-list
+
+   string-or-identifier:
+     string-literal
+     identifier
+
+The require statement takes a comma-separated list of param string
+filenames as operands. The filenames must be Chapel source files (*.chpl),
+C source files (*.c), C header files (*.h), precompiled C object
+files (*.o), or precompiled library archives (lib*.a). When using
+precompiled library archives, remove the lib and .a parts of the
+filename and add -l to the beginning as if it were being specified on
+the command line.
+
+.. code-block:: chapel
+
+   require "foo.h", "-lfoo";
+
+Require statements accept general ``param`` string expressions
+beyond the string literals shown in these examples.  Only
```

Comment:
Suggestion:

Each filename in the require statement must be given by a string literal or an identifier that is a `param` string expression, such as a `param` variable or a parenless function returning a param string.